substrate/runtime/primitives/src/traits.rs

```diff
@@ -147,10 +149,86 @@ impl<A: Executable, B: Executable> Executable for (A, B) {
 	}
 }
 
+/// Abstraction around hashing
+pub trait Hashing {
+	/// The hash type produced.
+	type Output;
+
+	/// Produce the hash of some byte-slice.
+	fn hash(s: &[u8]) -> Self::Output;
+
+	/// Produce the hash of some codec-encodable value.
+	fn hash_of<S: codec::Slicable>(s: &S) -> Self::Output {
+		codec::Slicable::using_encoded(s, Self::hash)
+	}
+
+	/// Produce the patricia-trie root of a mapping from indices to byte slices.
+	fn enumerated_trie_root(items: &[&[u8]]) -> Self::Output;
```

Comment:
this and other `trie_root` functions should be able to go away with upcoming generalization. Although it will use a slightly different `Hash` trait

Reply:
agree. one for a later refactoring pr though.